crm_compassion/models/event_compassion.py

```diff
@@ -151,10 +152,25 @@ class event_compassion(models.Model):
     parent_copy = fields.Many2one(
         'account.analytic.account', related='parent_id')
     calendar_event_id = fields.Many2one('calendar.event')
+    hold_start_date = fields.Date(
+        'Hold Start Date',
+        default=lambda self: self._default_hold_start_date(),
+        required=True
+    )
 
     ##########################################################################
     #                             FIELDS METHODS                             #
     ##########################################################################
+    @api.model
+    def _default_hold_start_date(self):
+        days_allocate_before_event = int(
+            self.env['ir.config_parameter'].get_param(
+                'crm_compassion.days_allocate_before_event'
+            )
+        )
+        self.hold_start_date = fields.date.today() - timedelta(

```

Comment:
`fields.date.today()` returns a string. So you should use `datetime.today()` instead to have a date object.
Moreover you should not assign the result to `self` in a `api.model` method. Return the result instead. It should also be converted to a string before returned (using `fields.Date.to_string()`)